client/src/main/java/org/apache/iotdb/client/AbstractClient.java

```diff
@@ -532,45 +539,61 @@ private static void executeQuery(IoTDBConnection connection, String cmd) {
     println(String.format("It costs %.3fs", costTime / 1000.0));
   }
 
+  private static void printNextBatchResult(ZoneId zoneId, ResultSet resultSet,
+      ResultSetMetaData resultSetMetaData, int columnLength, List<Integer> maxSizeList,
+      int preLinesCount)
+      throws SQLException {
+    maxSizeList.clear();
+    List<List<String>> lists = cacheResult(resultSet, maxSizeList, columnLength,
+        resultSetMetaData, zoneId);
+    output(lists, maxSizeList);
+    if (isReachMaxLine) {
+      println(String.format("Reach maxPrintRowCount = %s lines, enter q to exit, other to continue",
+          maxPrintRowCount));
+    }
+    printCount(preLinesCount + lists.get(0).size() - 1);
+  }
+
   /**
    * cache all results
-   * @param resultSet jdbc resultSet
-   * @param maxSizeList the longest result of every column
-   * @param columnCount the number of column
+   *
+   * @param resultSet         jdbc resultSet
+   * @param maxSizeList       the longest result of every column
+   * @param columnCount       the number of column
    * @param resultSetMetaData jdbc resultSetMetaData
-   * @param zoneId your time zone
-   * @return List<List<String>> result
+   * @param zoneId            your time zone
+   * @return List<List < String>> result
    * @throws SQLException throw exception
    */
   private static List<List<String>> cacheResult(ResultSet resultSet, List<Integer> maxSizeList,
       int columnCount, ResultSetMetaData resultSetMetaData, ZoneId zoneId)
       throws SQLException {
     boolean printTimestamp = !((IoTDBQueryResultSet) resultSet).isIgnoreTimeStamp();
     List<List<String>> lists = new ArrayList<>(columnCount);
-    for(int i = 1; i <= columnCount; i++) {
-      List<String> list = new ArrayList<>(1001);
```

Comment:
change it to 1001 or maxPrintRowCount + 1. It has one more line that is name header row.